library/tinkerpop/src/main/java/uk/gov/gchq/gaffer/tinkerpop/GafferPopGraph.java

```diff
@@ -799,70 +790,44 @@ private View createAllEntitiesView() {
         return viewBuilder.build();
     }
 
+    /**
+     * Determines the element seeds based on supplied IDs.
+     * Supports input as a {@link Vertex}, {@link Edge}, List of Edge IDs or individual Vertex IDs.
+     *
+     * @param ids The iterable of IDs
+     * @return List of {@link ElementSeed}s
+     */
     private List<ElementSeed> getElementSeeds(final Iterable<Object> ids) {
-        List<ElementSeed> seeds = null;
-        if (null != ids) {
-            seeds = new LinkedList<>();
-            for (final Object id : ids) {
-                if (id instanceof Vertex) {
-                    seeds.add(new EntitySeed(((Vertex) id).id()));
-                } else if (id instanceof GafferPopEdge) {
-                    seeds.add(new EdgeSeed(((GafferPopEdge) id).outVertex().id(), ((GafferPopEdge) id).inVertex().id(), true));
-                } else if (id instanceof List) {
-                    final List edgeIdList = (List) id;
-                    if (edgeIdList.size() == 2) {
-                        seeds.add(new EdgeSeed(edgeIdList.get(0), edgeIdList.get(1), true));
-                    }
-                } else {
-                    seeds.add(new EntitySeed(id));
-                }
-            }
+        List<ElementSeed> seeds = new LinkedList<>();
+        if (null == ids) {
+            LOGGER.warn("Gaffer does not support unseeded queries, no results will be returned");
+            return seeds;
         }
 
-        return seeds;
-    }
-
-    private List<EntitySeed> getEntitySeeds(final Iterable<Object> vertexIds) {
```

Comment:
What is the reason for moving these into one function? Isn't it technically more performant to have smaller functions? If you are in the `edges` function, all of your seeds should be edgeSeeds? Or is this assumption incorrect with tinkerpop?

Reply:
The assumption appears to be incorrect for tinkerpop as, the interface will request using the `edges()` method which tinkerpop assumes can also accept vertex ID objects as input so that edges on those given vertexes will be returned. Separating the functionality does not all for that use case.